ocs_ci/ocs/longevity.py

```diff
@@ -0,0 +1,704 @@
+                        thread3 = executor2.submit(cos.cosbench_full())
+                        threads.append(thread3)
+                        continue
+                    else:
+                        log.info(f"Starting {workload} workload in foreground")
+                        cos.cosbench_full()
+                        cos.cleanup()
+            if run_in_bg:
+                for t in threads:
+                    t.result()
+                cleanup()
+            log.info(f"##############COMPLETED CYCLE:{cycle_count}####################")
+            cycle_count += 1
+
+    def cleanup():
+        for workload in workloads:
+            try:
+                workload.cleanup()
+            except CommandFailed as ef:
+                log.info("Workload already cleaned")
+                if "does not exist on " not in str(ef):
+                    raise ef
+
+    request.addfinalizer(cleanup)
+    return factory
+
+
+def start_ocp_workload(workloads_list, run_in_bg=True):
+    """
+    This function reads the list of OCP workloads to run and
+     starts running those iterating over the elements in the list.
+
+     Usage:
+     start_ocp_workload(workloads_list=['logging','registry'], run_in_bg=True)
+
+     Args:
+         workloads_list (list): The list of ocp workloads to run
+         run_in_bg (bool): Runs the workload in background starting a thread
+
+    Raise:
+        UnsupportedWorkloadError: When the workload is not found in the supported_ocp_workloads list
+
+    """
+    threads = []
+    support_check = all(item in supported_ocp_workloads for item in workloads_list)
+    if not support_check:
+        raise UnsupportedWorkloadError("Found Unsupported ocp workloads list")
+
+    for workload in workloads_list:
+        if workload == "monitoring":
+            if not check_if_monitoring_stack_exists():
+                if run_in_bg:
+                    log.info(f"Starting {workload} workload in background")
+                    executor = ThreadPoolExecutor(max_workers=1)
+                    thread1 = executor.submit(setup_persistent_monitoring)
+                    threads.append(thread1)
+                    continue
+                else:
+                    log.info(f"Starting {workload} workload in foreground")
+                    setup_persistent_monitoring()
+        elif workload == "registry":
+            if not check_if_registry_stack_exists():
+                if run_in_bg:
+                    log.info(f"Starting {workload} workload in background")
+                    executor2 = ThreadPoolExecutor(max_workers=1)
+                    thread2 = executor2.submit(change_registry_backend_to_ocs)
+                    threads.append(thread2)
+                    continue
+                else:
+                    log.info(f"Starting {workload} workload in foreground")
+                    change_registry_backend_to_ocs()
+        elif workload == "logging":
+            if run_in_bg:
```

Comment:
Though least chances logging will be installed on our cluster would be better to validate is it already deployed or not, like others. WDYT?

Reply:
It is handled. Please check
https://github.com/red-hat-storage/ocs-ci/pull/5693/files#diff-040e1b82797a176d241b800cf5bfc312cbba7ce5c36863efa9796ffee3d39436R320-R327